### PR TITLE
Pddl generator

### PR DESCRIPTION
**Pull request by kris701**
Closes issue #12

### CODE REVIEW

File: src/PDDLCodeGenerator/PDDLCodeGenerator.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+bool PDDLCodeGenerator::VerifyCodeGeneration(string sourceFile, string generatedFile) {
+    ifstream sourceStream(sourceFile);
+    string sourceBuffer((   istreambuf_iterator<char>(sourceStream)),
+                            istreambuf_iterator<char>());
+    ifstream generatedStream(generatedFile);
+    string generatedBuffer((istreambuf_iterator<char>(generatedStream)),
+                            istreambuf_iterator<char>());
+
+    sourceBuffer.erase(remove(sourceBuffer.begin(), sourceBuffer.end(), '\n'), sourceBuffer.end());
```
Comment (jamadaha) [inline src/PDDLCodeGenerator/PDDLCodeGenerator.cpp:26]: All of these whitespace removals can be replaced by 
`sourceBuffer.erase(remove_if(sourceBuffer.begin(), sourceBuffer.end(), ::isspace), sourceBuffer.end());`
If you include <cctype>

File: src/PDDLCodeGenerator/PDDLDomainCodeGenerator.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
+#include "PDDLDomainCodeGenerator.h"
+
+using namespace std;
+
+void PDDLDomainCodeGenerator::GenerateDomainFile(Domain* domain, string domainFile) {
+	ofstream file;
+	file.open(domainFile);
+	file << "(define (domain " + domain->_name + ")" << endl;
```
Comment (jamadaha) [inline src/PDDLCodeGenerator/PDDLDomainCodeGenerator.cpp:8]: Why so many endl? Why not \n?

File: src/PDDLCodeGenerator/PDDLProblemCodeGenerator.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#include "PDDLProblemCodeGenerator.h"
+
+using namespace std;
+
+void PDDLProblemCodeGenerator::GenerateProblemFile(Problem* problem, string problemFile) {
+	ofstream file;
+	file.open(problemFile);
+	file << "(define (problem " + problem->_name + ")" << endl;
```
Comment (jamadaha) [inline src/PDDLCodeGenerator/PDDLProblemCodeGenerator.cpp:8]: Again why use endl?

**Review by jamadaha — CHANGES_REQUESTED** [review]

File: src/PDDLCodeGenerator/PDDLCodeGenerator.cpp (hunk 1 above)
Comment (kris701) [reply src/PDDLCodeGenerator/PDDLCodeGenerator.cpp]: Just refactored it instead

**Review by jamadaha — APPROVED** [review]

**Pull request merged**